static/js/settings_user_groups.js

```diff
@@ -32,19 +33,13 @@ export function reload() {
     populate_user_groups();
 }
 
-const USER_GROUP_EDIT_POLICY_MEMBERS = 1;
-
 export function can_edit(group_id) {
-    if (page_params.is_admin) {
-        return true;
-    }
-
-    if (page_params.is_guest) {
+    if (!settings_data.user_can_edit_user_groups()) {
         return false;
     }
 
-    if (page_params.realm_user_group_edit_policy !== USER_GROUP_EDIT_POLICY_MEMBERS) {
-        return false;
+    if (page_params.is_admin || page_params.is_moderator) {
+        return true;
     }
```

Comment:
Can you add a comment above this block explaining that moderators and above are allowed to edit any user group, whereas normal users are only allowed to edit groups they are a member of (if given permission at all).